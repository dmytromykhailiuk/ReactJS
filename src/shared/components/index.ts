export * from "./Footer";
export * from "./Logo";
export * from "./Button";
export * from "./Input";
export * from "./ThreeDotsIcon";
export * from "./ErrorBoundary";
