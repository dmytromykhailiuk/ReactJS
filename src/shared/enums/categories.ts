export enum Categories {
  ALL = "ALL",
}
