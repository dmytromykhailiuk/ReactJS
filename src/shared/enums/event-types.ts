export enum EventTypes {
  CLICK = "click",
  SCROLL = "scroll",
}
