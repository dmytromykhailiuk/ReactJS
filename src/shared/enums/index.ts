export * from "./button-types";
export * from "./button-size";
export * from "./categories";
export * from "./sorting-options";
export * from "./movie-form-values";
export * from "./router-paths";
export * from "./modal-types";
