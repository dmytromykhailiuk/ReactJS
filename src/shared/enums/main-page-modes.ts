export enum MainPageModes {
  OVERVIEW = "overview",
  EDIT = "edit",
  CREATE = "create",
  DELETE = "delete",
}
