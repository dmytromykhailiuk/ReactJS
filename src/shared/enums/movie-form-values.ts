export enum MovieFormValues {
  MOVIE_ID = "id",
  TITLE = "title",
  RELEASE_DATE = "releaseDate",
  MOVIE_URL = "url",
  CATEGORY = "category",
  OVERVIEW = "overview",
  RUNTIME = "runtime",
  DURATION = "duration",
  RATING = "rating",
}
