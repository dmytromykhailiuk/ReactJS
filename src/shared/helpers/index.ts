export * from "./filterMovies";
export * from "./sortMovies";
export * from "./generateId";
export * from "./clickOutside";
