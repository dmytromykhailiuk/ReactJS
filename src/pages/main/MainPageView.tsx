import React, { useMemo } from "react";
import { Banner, MovieBoard } from "./components";
import { renderModal } from "shared/helpers"
import { Movie } from "models/";
import { ModalTypes, RouterPaths } from "shared/enums";
import { SearchPanel, MovieItemDetails } from "./components";
import { Switch, Route } from "react-router";

const bluredStyles = {
  filter: 'blur(7px)',
}

export interface MainPageViewProps {
  modalInView: ModalTypes;
  hasMovieInOverview: boolean;
  selectedMovie: Movie;
  movieBoard: React.MutableRefObject<any>;
  movies: Movie[];
  isSuccessAlert: boolean;
  alertMessage: string;
  navigateToMainPage: () => void;
  onCreateMovie: () => void;
  navigateToSearchPage: () => void;
  setSearchingValue: (searchingValue: string) => void;
  onEditMovie: (movie: Movie) => void;
  onDeleteMovie: (movie: Movie) => void;
  onCloseModal: () => void;
  onCloseWithSaving: (movie: Movie) => void;
}

const MainPageView: React.FC<MainPageViewProps> = ({ 
  modalInView, 
  hasMovieInOverview, 
  movieBoard,
  movies, 
  isSuccessAlert, 
  selectedMovie, 
  alertMessage, 
  navigateToMainPage, 
  onCreateMovie, 
  navigateToSearchPage, 
  setSearchingValue, 
  onEditMovie, 
  onDeleteMovie, 
  onCloseModal, 
  onCloseWithSaving
}) => {

  const Modal = useMemo(() => renderModal(modalInView), [modalInView]);
  
  const contentStyles = modalInView ? bluredStyles: {};

  return (
    <>
      <header style={contentStyles}>
        <Banner
          isMovieInOverviewMode={hasMovieInOverview}
          onCreateMovie={onCreateMovie}
          onLogoClicked={navigateToMainPage}
          onSearchIconClicked={navigateToSearchPage}
        >
          <Switch>
            <Route 
              path={RouterPaths.FILM} 
              component={MovieItemDetails} 
            />
            <Route path={[RouterPaths.HOME, RouterPaths.SEARCH]} exact>
              <SearchPanel onChangeSearchingValue={setSearchingValue} />
            </Route>
          </Switch>
        </Banner>
      </header>

      <main style={contentStyles}>
        <MovieBoard
          movieBoardRef={movieBoard}
          movies={movies}
          onEditMovie={onEditMovie}
          onDeleteMovie={onDeleteMovie}
        />
      </main>

      { modalInView && (
        <Modal 
          movie={selectedMovie}
          isSuccessAlert={isSuccessAlert}
          alertMessage={alertMessage}
          onCloseModal={onCloseModal}
          onCloseWithSaving={onCloseWithSaving}
        />) }
    </>
  )
}

export default MainPageView;
