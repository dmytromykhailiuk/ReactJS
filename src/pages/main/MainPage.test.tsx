import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import toJson from 'enzyme-to-json';
import MainPageView from "./MainPageView";
import { ModalTypes } from "shared/enums";
import { renderModal } from "shared/helpers";

describe("MainPageView", () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    wrapper = shallow(
      <MainPageView
        movies={[{ id: 2 }] as any[]}
        movieBoard={{} as any}
        isSuccessAlert={true}
        modalInView={null}
        hasMovieInOverview={false}
        selectedMovie={null}
        alertMessage={"alertMessage"}
        onEditMovie={() => {}}
        onDeleteMovie={() => {}}
        setSearchingValue={() => {}}
        navigateToSearchPage={() => {}}
        onCloseWithSaving={() => {}}
        onCloseModal={() => {}}
        onCreateMovie={() => {}}
        navigateToMainPage={() => {}}
      />
    )
  })

  it("should match snepshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it("should show modal", () => {
    wrapper.setProps({ modalInView: ModalTypes.DELETE })
    expect(wrapper.exists(renderModal(ModalTypes.DELETE))).toBeTruthy();
  })
});
