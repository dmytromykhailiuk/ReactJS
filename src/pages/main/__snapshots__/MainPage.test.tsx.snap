// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPageView should match snepshot 1`] = `
<Fragment>
  <header
    style={Object {}}
  >
    <Memo()
      isMovieInOverviewMode={false}
      onCreateMovie={[Function]}
      onLogoClicked={[Function]}
      onSearchIconClicked={[Function]}
    >
      <Switch>
        <Route
          component={[Function]}
          path="/film/:id"
        />
        <Route
          exact={true}
          path={
            Array [
              "/",
              "/search",
            ]
          }
        >
          <SearchPanel
            onChangeSearchingValue={[Function]}
          />
        </Route>
      </Switch>
    </Memo()>
  </header>
  <main
    style={Object {}}
  >
    <Component
      movieBoardRef={Object {}}
      movies={
        Array [
          Object {
            "id": 2,
          },
        ]
      }
      onDeleteMovie={[Function]}
      onEditMovie={[Function]}
    />
  </main>
</Fragment>
`;
